src/reducers/ComparisonSlice.ts

```diff
@@ -18,6 +19,15 @@ type Results = {
   revisionHeader: RevisionsHeader;
 };
 
+interface GroupedResults {
+  [key: string]: ResultObject[];
+}
```

Comment:
This is just like `Record<string, ResultObject[]>`, so you could do:

```js
type ResultsGroupedByKey = Record<string, ResultObject[]>
```
And use `ResultsGroupedByKey` everywhere. Also I suggest to change the type name here, so that we understand better what's inside.